src/app/components/week-selector/week-selector.component.ts

```diff
@@ -0,0 +1,69 @@
+import { async } from '@angular/core/testing';
+import { Component, OnInit, Input, Output, EventEmitter, AfterContentChecked } from '@angular/core';
+import { addWeeks, subWeeks, subYears, getWeek, getYear } from 'date-fns';
+
+@Component({
+  selector: 'app-week-selector',
+  templateUrl: './week-selector.component.html',
+  styleUrls: ['./week-selector.component.scss']
+})
+export class WeekSelectorComponent implements OnInit {
+
+  @Input() currentDate: Date;
+  @Input() startDate: Date;
+  @Input() subTitle: string;
+
+  @Output() passDate = new EventEmitter<number>();
+
+  weekNumber: number;
+  year: number;
+  isCurrentWeek: boolean;
+  isStartingWeek: boolean;
+  disabledLeft: boolean = false;
+  disabledRight: boolean;
+  date: Date
+
+  constructor() {}
+
+  ngOnInit() {
+    this.isCurrentWeek = true;
+    this.disabledRight = true;
+    this.date = this.currentDate
+    this.weekNumber = getWeek(this.date);
+    this.passDate.emit(this.weekNumber)
+    this.year = getYear(this.date)
+  }
+
+  onPreviousWeek() {
+    this.date = subWeeks(this.date, 1);
+    this.weekNumber = getWeek(this.date)
+    
+    this.setCurrentWeekBoolean();
+    
+    
+    this.disabledRight = false;
+    this.disabledLeft = this.isDateBeforeStartDate();
+    
+    this.passDate.emit(this.weekNumber);
+  }
+
+  onNextWeek() {
+    this.date = addWeeks(this.date, 1);
+    this.weekNumber = getWeek(this.date)
+    
+    this.passDate.emit(this.weekNumber);
+    
+    this.disabledLeft = false;
```

Comment:
Wil je deze niet dynamisch maken oid, in plaats van handmatig te setten? Ik zie dat je ze wel hebt getest, dus het is wel redelijk robuust, maar alsnog. Ik zou in de template denk ik een method aanroepen die het dynamisch ophaalt. Dat maakt de code wel duidelijk, denk ik

Reply:
Ja aan de ene kant heb je wel gelijk. Het is wel zo dat deze na het voeren van de functie nooit op true zou moten kunnen staan.

Reply:
Toch maar een methode aan assigned